mypy/typeshed/stdlib/typing.pyi

```diff
@@ -902,10 +902,10 @@ if sys.version_info >= (3, 9):
     @overload
     def get_origin(tp: GenericAlias) -> type: ...
     @overload
-    def get_origin(tp: Any) -> Any | None: ...
+    def get_origin(tp: Any) -> object: ...
 
 else:
-    def get_origin(tp: Any) -> Any | None: ...
+    def get_origin(tp: Any) -> object: ...
```

Comment:
would it be better to be `object & HasBool`?

Reply:
do you actually support `python < 3.9`?

Reply:
yes, 3.8 is currently supported

Reply:
> `object & Hasbool` is equivalent to `HasBool`, right?

Oh wait, no, `object.__bool__` isn't a thing. I forgot that `bool()` just returns `True` for objects that have `__bool__` or `__len__` (https://docs.python.org/3/library/stdtypes.html#truth-value-testing)

Reply:
`object | None` then?